### PR TITLE
Add pre-filter for files in setrpaths.sh for ELF/wheels/jars.

### PR DESCRIPTION
**Pull request by bartoldeman**
All ELF files have a 4 byte header '\x7fELF'. By scanning files for the filename and this header only we can disregard all regular files more quickly, and only let "file -b" do the heavy lifting on the smaller set.

This speeds up setrpaths.sh significantly: e.g. on a Narval SSD on a MATLAB repo ( > 500,000 files) it went from 2h31m49s to 5m35s.

### CODE REVIEW

File: setrpaths.sh [hunk 1]
```diff
@@ -134,7 +134,19 @@ fi
 
 SAVEIFS=$IFS
 IFS=$(echo -en "\n\b")
-for filename in $(find $ARG_PATH -type f); do
+for filename in $(find $ARG_PATH -type f -print0 | python -c '
```
Comment (mboisson) [inline setrpaths.sh:137]: could filter `.whl` and `.jar` at the `find` level instead of in python, with `-not -name '*.whl' -a -not -name '*.jar'`

Comment (mboisson) [reply setrpaths.sh:137]: we could also speed it up by excluding other file types, like `.html`, `.txt`, etc.

Comment (bartoldeman) [reply setrpaths.sh:137]: hmm we want to specifically *include* whl and jar files and no look at them, not exclude them.
I can however restructure it with the Python code returning an exit code and `find` testing for that.

excluding by extension can be a bit tricky, e.g. for MATLAB most end in `.m` but there may be some other program that has executables ending in `.m`, so I'd rather not go down that rabbit hole.

Comment (mboisson) [reply setrpaths.sh:137]: ah, right, it would require two finds (one for jar/whl and one for the rest). Disregard my comment.

**Comment by mboisson** [comment]
lgtm

**Pull request merged**